Document comparison of Alert Suppression, Snoozing, Exceptions, Maintenance Windows (#3314)
elastic/security-docs: docs/detections/alerts-reduce.asciidoc

[[reduce-notifications-alerts]]
== Reduce notifications and alerts

> {elastic-sec} offers several features to help reduce the amount of notifications and alerts produced by your detection rules. This table provides a general comparison of these features, with links for more details:

Reviewer: ```suggestion
{elastic-sec} offers several features to help reduce the amount of notifications and alerts generated by your detection rules. This table provides a general comparison of these features, with links for more details:
```
Either way works but I usually see "alert generation" 🤷

> {elastic-sec} offers several features to help reduce the amount of notifications and alerts generated by your detection rules. This table provides a general comparison of these features, with links for more details:

Reviewer 2: Slight nit here for `number` vs `amount`. 
```suggestion
{elastic-sec} offers several features to help reduce the number of notifications and alerts generated by your detection rules. This table provides a general comparison of these features, with links for more details:
```

[cols="2"]
|===

| <<snooze-rule-actions,Rule action snoozing>>

> a| *_Prevents a specific rule's notification actions from running_*. 

Reviewer 3: I think "stop" is the more accurate verb because "prevent" seems more preemptive. I could be wrong, but I don't think users would proactively snooze action notifications after enabling a rule because they thought the rule would be noisy. 

```suggestion
a| *_Stops a specific rule's notification actions from running_*. 
```

> Use to avoid unnecessary notifications from a noisy rule. The rule continues to run and generate alerts during the snooze period, but its <<rule-notifications,notification actions>> are suppressed.

Reviewer 3: While this can be used to quiet a noisy rule, it can also be used to temporarily stop notifications from being sent during certain timeframes (e.g., during application/network maintaince).

```suggestion
Use to avoid unnecessary notifications from specific rules. The rule continues to run and generate alerts during the snooze period, but its <<rule-notifications,notification actions>> are suppressed.
```

Author: Good call, let's actually include all those as examples:
```suggestion
Use to avoid unnecessary notifications from a specific rule. For example, you can turn off a rule's notifications during application or network maintenance. The rule continues to run and generate alerts during the snooze period, but its <<rule-notifications,notification actions>> are suppressed.
```

Author: ```suggestion
Use to avoid unnecessary notifications from a specific rule. The rule continues to run and generate alerts during the snooze period, but its <<rule-notifications,notification actions>> are suppressed.
```

| {kibana-ref}/maintenance-windows.html[Maintenance window]
a| *_Prevents all rules' notification actions from running_*. 

> Use to avoid false alarms and unnecessary notifications during planned outages. All rules continue to run and generate alerts during the maintenance window, but their <<rule-notifications,notification actions>> are suppressed.

Reviewer 3: Making this suggestion in case we want to avoid over-using "suppression".

```suggestion
Use to avoid false alarms and unnecessary notifications during planned outages. All rules continue to run and generate alerts during the maintenance window, but their <<rule-notifications,notification actions>> are stopped.
```

> NOTE: Maintenance windows are a {kib} feature and are configured outside of the {security-app} in *Stack Management*.

Reviewer 2: ```suggestion
NOTE: Maintenance windows are a {kib} feature, configured outside of the {security-app} in *Stack Management*.
```

| <<alert-suppression,Alert suppression>>
a| *_Reduces repeated or duplicate alerts created by a custom query rule_*. 

Use to reduce the number of alerts created by a <<create-custom-rule,custom query rule>> that matches multiple source events. Matching events are grouped by their values in a specified field, and only one alert is created for each group.

| <<detections-ui-exceptions,Rule exception>>
a| *_Prevents a rule from creating alerts under specific conditions_*.

> Use to reduce false positive alerts, by preventing trusted processes and network activity from generating unnecessary alerts. You can configure an exception to be used by a single rule or shared among multiple rules, but they typically don't affect _all_ rules.

Reviewer 2: ```suggestion
Use to reduce false positive alerts by preventing trusted processes and network activity from generating unnecessary alerts. You can configure an exception to be used by a single rule or shared among multiple rules, but they typically don't affect _all_ rules.
```
